poppy/poppy_core.py

```diff
@@ -2158,17 +2158,15 @@ def __init__(self, name="unnamed optic", transmission=None, opd= None, opdunits=
                     _log.error("No opdunit keyword supplied, and BUNIT keyword not found in header. Cannot determine OPD units")
                     raise Exception("No opdunit keyword supplied, and BUNIT keyword not found in header. Cannot determine OPD units.")
 
-
-            if opdunits.lower().endswith('s'): opdunits = opdunits[:-1] # drop trailing s if present
-            if opdunits.lower() == 'meter' or opdunits.lower() == 'm':
-                pass # no need to rescale
-            elif opdunits.lower() == 'micron' or opdunits.lower() == 'um' or opdunits.lower() == 'micrometer':
+            opdunits = opdunits.lower()
+            # rescale to meters if necessary
+            if opdunits in ('meter', 'meters', 'm'):
+                pass
+            elif opdunits in ('micron', 'microns', 'um', 'micrometer', 'micrometers'):
```

Comment:
OK, that's fine. This is micro-optimizing something that's not at all computationally demanding :-) I was mostly just idly curious since I couldn't figure out any technical need for this associated with the BUNIT changes.